packages/mongodb-log-writer/src/mongo-log-manager.ts

```diff
@@ -94,9 +94,17 @@ export class MongoLogManager {
           fullPath,
         };
       } else if (this._options.retentionGB || this._options.maxLogFileCount) {
-        const fileSize = (await fs.stat(fullPath)).size;
+        let fileSize: number | undefined;
+
         if (this._options.retentionGB) {
-          usedStorageSize += fileSize;
+          try {
+            fileSize = (await fs.stat(fullPath)).size;
+            if (this._options.retentionGB) {
+              usedStorageSize += fileSize;
+            }
+          } catch (err) {
+            this._options.onerror(err as Error, fullPath);
```

Comment:
is this reasonable behavior or should we cancel the cleanup altogether in this case?

Reply:
It seems very reasonable to me 👍